src/legacy/work/Header/Header.tsx

```diff
@@ -68,33 +71,58 @@ const DetailedHeader = () => {
         <NavbarToggler onClick={handleToggleNavbar} className="mr-2" />
         <Collapse isOpen={isOpen} navbar>
           <Nav className="ml-auto u-header__navbar-nav pt-5 pb-3 pt-md-0 pb-md-0" navbar>
-            <Link to={HOST_INTEREST_LINK} className="mb-4 mb-md-0 mr-md-4 w-100 w-md-auto btn btn-outline-primary">
-              Become a Host
-            </Link>
             <FirebaseConsumer>
               {({ loading, user }: FirebaseUserProps) => {
                 if (loading) {
                   return <Loading />;
                 }
                 if (user) {
                   return (
-                    authNavItems.map(item => (
+                    <Query query={GET_USER}>
+                      {({ loading, error, data }) => {
+                        if (loading) {
+                          return <Loading />;
+                        }
+                        if (error || !data) {
+                          return <h1>{error ? error.message : 'Error / No Data'}</h1>;
+                        }
+                        const beeUser = data.user;
+                        return (
+                          <>
+                            {beeUser.listingCount > 0
+                              ? <Link to={HOST_PORTAL_LINK} className="mb-4 mb-md-0 mr-md-4 w-100 w-md-auto btn btn-outline-primary">
+                                  Host Profile
+                                </Link>
+                              : <Link to={HOST_INTEREST_LINK} className="mb-4 mb-md-0 mr-md-4 w-100 w-md-auto btn btn-outline-primary">
+                                  Become a Host
+                                </Link>
+                            } 
+                            {authNavItems.map(item => (
+                              <NavItem className="px-2" key={item.header}>
+                                <NavLink to={item.link} tag={Link}>
+                                  {item.header}
+                                </NavLink>
+                              </NavItem>
+                            ))}
+                          </>
+                        );
+                      }}
+                    </Query>
+                  )
+                }
```

Comment:
All this just to dynamically render the Host Portal button name on the header O_O

Reply:
🙈

Reply:
Worth moving some of this into its own component? The nesting gets pretty deep, but 142 LOC still seems reasonable for the Header as a whole, so I'm neutral

Reply:
Hmmm nesting does get pretty deep. I don't anticipate the Header changing very much from this point so I think it'd be good to keep logic local for now.